sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/FileIndexer.java

```diff
@@ -70,29 +69,39 @@
   private final DefaultInputModule module;
   private final BatchIdGenerator batchIdGenerator;
   private final InputComponentStore componentStore;
+  private final ModuleFileSystemInitializer moduleFileSystemInitializer;
   private ExecutorService executorService;
   private final List<Future<Void>> tasks;
+  private final DefaultModuleFileSystem defaultModuleFileSystem;
+  private final LanguageDetection langDetection;
 
   private ProgressReport progressReport;
 
   public FileIndexer(BatchIdGenerator batchIdGenerator, InputComponentStore componentStore, DefaultInputModule module, ExclusionFilters exclusionFilters,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 10 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)